lua/plenary/job.lua

```diff
@@ -370,13 +370,16 @@ function Job:_execute()
     if Job.is_job(self.writer) then
       self.writer:_execute()
     elseif type(self.writer) == 'table' and vim.tbl_islist(self.writer) then
-      for _, v in ipairs(self.writer) do
-        self.stdin:write(v .. '\n')
-      end
-      self.stdin:close()
+      local writer = vim.tbl_map(function(l)
+        return l..'\n'
+      end, self.writer)
+      self.stdin:write(writer, function()
+        self.stdin:close()
```

Comment:
for this case, I don't want to concat the table.

Instead I'd like to loop through and writ e them, and then we can check if the index in `ipairs` is equal to the length of `self.writer`. If so, then we can do the close check.

Does that make sense?

Also, we can change it to be `self.stdin:write(v)` and then `self.stdin:write("\n")`, rather than concating them. String concat is not great in lua (and I'm trying to be more picky about the perf of `job`).